code/obj/machinery/rat_den.dm

```diff
@@ -0,0 +1,45 @@
+/obj/machinery/wraith/rat_den
+	name = "rat den"
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "rat_den"
+	desc = "A pile of garbage vaguely ressembling a nest."
+	anchored = 1
+	density = 0
+	_health = 25
+	var/linked_critters = 0
+	var/max_critters = 5
+	var/next_spawn_check = 10 SECONDS
+
+	New()
+		..()
+		next_spawn_check = TIME + (10 SECONDS)
+		return
+
+	attackby(obj/item/P, mob/living/user)
+		src._health -= P.force
+		attack_particle(user,src)
+		user.lastattacked = src
+		hit_twitch(src)
+		playsound(src.loc, 'sound/impact_sounds/Metal_Hit_Light_1.ogg', 50, 1)
+		if(prob(20))
+			playsound(src.loc, 'sound/voice/animal/mouse_squeak.ogg', 60, 1)
+		if(src._health <= 0)
+			src.gib(src.loc)
+			qdel(src)
+
+	process()	//Spawn some mad mice once in awhile
+		if (src.next_spawn_check != null)
+			if (src.next_spawn_check < TIME)
+				next_spawn_check = TIME + rand(20 SECONDS, 25 SECONDS)
+				if (linked_critters < max_critters)
+					var/obj/critter/mouse/mad/ratden/M = new /obj/critter/mouse/mad/ratden(src.loc)
+					M.linked_den = src
+					linked_critters ++
+
+		//Plague rats in range heal up slowly
+		for (var/mob/living/critter/wraith/plaguerat/P in range(5, src))
+			if((P.health < (P.health_brute + P.health_burn)))
+				for(var/damage_type in P.healthlist)
+					var/datum/healthHolder/hh = P.healthlist[damage_type]
+					hh.HealDamage(3)
+				boutput(P, "The proximity of the rat den fills you with renewed malevolence.")
```

Comment:
tons of 5+ range checks in this PR - stuff like this should really be a by_type[] and dist check - would say spatial hashing but that's only clients